src/main/java/io/druid/data/input/impl/DelimitedParseSpec.java

```diff
@@ -33,7 +33,7 @@ public DelimitedParseSpec(
     this.delimiter = delimiter;
     this.listDelimiter = listDelimiter;
 
-    this.columns = columns == null ? Lists.<String>newArrayList() : columns;
+    this.columns = columns == null ? dimensionsSpec.getDimensions() : columns;
```

Comment:
metrics are not available at this place, so we cannot assume reasonable default, I think it will be better to not allow nulls as we do in CSVParseSpec since we cannot figure out reasonable defaults here..